modules/rtqueueview.php

```diff
@@ -72,23 +72,23 @@
 
 // category id's
 if (isset($_GET['catid'])) {
-    if (is_array($_GET['catid'])) {
-        if (in_array(-1, $_GET['catid'])) {
-            $filter['catids'] = -1;
-        } else {
-            $filter['catids'] = Utils::filterIntegers($_GET['catid']);
-        }
-    } elseif (intval($_GET['catid'])) {
-        $filter['catids'] = Utils::filterIntegers(array($_GET['catid']));
-    } elseif ($_GET['catid'] == 'all') {
-        $filter['catids'] = 0;
+    if (!is_array($_GET['catid'])) {
+        $_GET['catid'] = array($_GET['catid']);
+    }
+
+    if ($_GET['catids'] == 'all') {
+        $_GET['catids'] = array(0);
+    } else {
+        $filter['catids'] = Utils::filterIntegers($_GET['catid']);
     }
```

Comment:
Tu może wystąpić scenariusz w którym $_GET['catid'] nie jest array(), a robisz Utils::filterIntegers().

Reply:
Aaaa już widzę - najpierw zabezpieczasz się przed taką sytuacją, że $_GET['catid'] nie jest array() i wtedy robisz z niego array().
Co to ma robić konkretnego?

Reply:
Możliwość zaznaczenia opcji bez kategorii i dowolnej kategorii podczas filtrowania.

Reply:
W sytuacji, gdy podasz `&catid=all`:
https://github.com/lmsgit/lms/pull/1679/files#diff-876c63174db1e4a0ccd729a7cdb856e8R75-R77
`$_GET['catid']` uzyska zawartość `array('all')`,
a potem robisz test:
https://github.com/lmsgit/lms/pull/1679/files#diff-876c63174db1e4a0ccd729a7cdb856e8R79-R80
który nigdy nie będzie spełniony.

Reply:
Masz rację zapomniałem o ujęciu w kodzie parametru "all". Wydaje mi się, że wprowadzone poprawki w ostatnich 2 commitach rozwiązują problem.